server/comms/server.go

```diff
@@ -448,6 +449,25 @@ func (s *Server) Run(ctx context.Context) {
 		rr.Use(s.limitRate)
 		rr.Get("/config", routeHandler(msgjson.ConfigRoute))
 		rr.Get("/spots", routeHandler(msgjson.SpotsRoute))
+		rr.Get("/logo", func(w http.ResponseWriter, r *http.Request) {
+			handler := httpRoutes[msgjson.LogoRoute]
+			if handler == nil {
+				writeServerError(w, map[string]string{"error": "no handler"})
+				return
+			}
+
+			logo, err := handler(r.Context().Value(ctxThing))
+			if err != nil { // err will always be nil, but this might change.
+				writeServerError(w, map[string]string{"error": err.Error()})
+				return
+			}
+
+			logoBytes := logo.([]byte)
+			w.WriteHeader(http.StatusOK)
+			w.Header().Set("Content-Type", http.DetectContentType(logoBytes))
+			w.Header().Set("Content-Length", fmt.Sprintf("%d", len(logoBytes)))
+			w.Write(logoBytes)
```

Comment:
- WriteHeader cannot come before w.Header().Set() https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/net/http/server.go;l=100-102
- WriteHeader isn't even needed because Write will do it for you https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/net/http/server.go;l=120-121